ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiPartyManagementService.scala

```diff
@@ -69,52 +72,59 @@ private[apiserver] final class ApiPartyManagementService private (
     PartyDetails(details.party, details.displayName.getOrElse(""), details.isLocal)
 
   override def getParties(request: GetPartiesRequest): Future[GetPartiesResponse] =
-    partyManagementService
-      .getParties(request.parties.map(Ref.Party.assertFromString))
-      .map(ps => GetPartiesResponse(ps.map(mapPartyDetails)))
-      .andThen(logger.logErrorsOnCall[GetPartiesResponse])
+    withEnrichedLoggingContext(logging.parties(request.parties)) { implicit loggingContext =>
+      logger.info("Getting parties")
+      partyManagementService
+        .getParties(request.parties.map(Ref.Party.assertFromString))
+        .map(ps => GetPartiesResponse(ps.map(mapPartyDetails)))
+        .andThen(logger.logErrorsOnCall[GetPartiesResponse])
+    }
 
   override def listKnownParties(
       request: ListKnownPartiesRequest
-  ): Future[ListKnownPartiesResponse] =
+  ): Future[ListKnownPartiesResponse] = {
+    logger.info("Listing known parties")
     partyManagementService
       .listKnownParties()
       .map(ps => ListKnownPartiesResponse(ps.map(mapPartyDetails)))
       .andThen(logger.logErrorsOnCall[ListKnownPartiesResponse])
+  }
 
-  override def allocateParty(request: AllocatePartyRequest): Future[AllocatePartyResponse] = {
-
-    val validatedPartyIdentifier =
-      if (request.partyIdHint.isEmpty) {
-        Future.successful(None)
-      } else {
-        Ref.Party
-          .fromString(request.partyIdHint)
-          .fold(
-            error => Future.failed(ErrorFactories.invalidArgument(error)),
-            party => Future.successful(Some(party)),
-          )
-      }
-
-    validatedPartyIdentifier
-      .flatMap(party => {
-        val displayName = if (request.displayName.isEmpty) None else Some(request.displayName)
-        synchronousResponse
-          .submitAndWait(CreateSubmissionId.withPrefix(party), (party, displayName))
-          .map { case PartyEntry.AllocationAccepted(_, partyDetails) =>
-            AllocatePartyResponse(
-              Some(
-                PartyDetails(
-                  partyDetails.party,
-                  partyDetails.displayName.getOrElse(""),
-                  partyDetails.isLocal,
+  override def allocateParty(request: AllocatePartyRequest): Future[AllocatePartyResponse] =
+    withEnrichedLoggingContext(logging.party(request.displayName)) { implicit loggingContext =>
```

Comment:
Would it make more sense to rather have the `partyIdHint` here? Even though both are optional, the `displayName` has more relaxed uniqueness requirements and it's probably less useful to track information when debugging.

Reply:
Fixed.